### PR TITLE
LWDEV-10443 Implement application cache to store hourly trade volume per wallet per asset

### PR DESCRIPTION
**Pull request by olpapchenko**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,13 +18,15 @@ repositories {
...
 	maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
 }
+ext.kotlin_version = '1.2.10'
 
 ext {
 	springfox_swagger2 = "2.9.2"
 	springfox_swagger_ui = "2.9.2"
 	lmax_disruptor_version = '3.4.2'
-	gson_version = '2.6.2'
+	jedis_version = '2.9.0'
```
Comment (barsukovskii) [inline build.gradle]: wrong property?

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/entity/impl/TradeVolumeCacheImpl.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                .getOrPut(clientIdAssetId) { TreeSet() }
+    }
+
+    private data class Volume(val tradeIdx: Int,
+                         val timestamp: Date,
+                         val volume: BigDecimal,
+                         val clientId: String,
+                         val assetId: String) : Comparable<Volume> {
+        override fun compareTo(other: Volume): Int {
+            return Comparator
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/trade/volume/monitoring/service/entity/impl/TradeVolumeCacheImpl.kt:130]: The comparator looks wrong:
1. second `reversed()` must be applied to `Comparator.comparingInt`
2. compare(other, this) -> compare(this, other)

What agreement about tradeIdx to pass it to the cache ? Just index of got trades from ME ?

**Review by barsukovskii — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (olpapchenko) [reply build.gradle]: removed

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/entity/impl/TradeVolumeCacheImpl.kt (hunk 2 above)
Comment (olpapchenko) [reply src/main/kotlin/com/lykke/trade/volume/monitoring/service/entity/impl/TradeVolumeCacheImpl.kt:130]: fixed, yes - trade id is `tradeIdx` is trade `index` from ME

**Review by barsukovskii — APPROVED** [review]

**Pull request merged**